api/src/main/java/org/pmiops/workbench/api/util/SQLGenerator.java

```diff
@@ -0,0 +1,166 @@
+package org.pmiops.workbench.api.util;
+
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.QueryRequest;
+import org.pmiops.workbench.model.SearchParameter;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class SQLGenerator {
+
+    private static final Map<String, String> typeCM = new HashMap<>();
+    private static final Map<String, String> typeProc = new HashMap<>();
+    static {
+        typeCM.put("ICD9", "ICD9CM");
+        typeCM.put("ICD10", "ICD10CM");
+        typeCM.put("CPT", "CPT4");
+
+        typeProc.put("ICD9", "ICD9Proc");
+        typeProc.put("ICD10", "ICD10PCS");
+        typeProc.put("CPT", "CPT4");
+    }
+
+    public QueryRequest findGroupCodes(String type, List<String> codes) {
+        StringBuilder queryBuilder = new StringBuilder();
+        queryBuilder.append("SELECT code, domain_id AS domainId FROM ");
+        queryBuilder.append(String.format("`%s.%s_criteria` ", getTablePrefix(), type.toLowerCase()));
+        queryBuilder.append("WHERE (");
+
+        Map<String, QueryParameterValue> queryParams = new HashMap<>();
+        List<String> queryParts = new ArrayList<>();
+        for (String code: codes) {
+            queryParams.put(code, QueryParameterValue.string(code));
+            queryParts.add("code LIKE @"+code);
```

Comment:
Why LIKE here? Is this doing a substring match?

BQ identifiers must follow this syntax:

https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#identifiers

Are we sure all values of code will match that? If not, might be better to call these something else... code1, code2, etc.